parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -1019,119 +1018,53 @@ static bool cxxParserParseNextTokenSkipIgnoredParenthesis(CXXToken ** ppChain)
-	CXX_DEBUG_ASSERT(pInfo->replacementLength > 0,"The replacement is too short");
+	CXX_DEBUG_ASSERT(pInfo->replacements,"There should be a replacement");
 
-	if(!pInfo->ignoreFollowingParenthesis)
+	if(!pInfo->hasParameterList)
 	{
 		CXX_DEBUG_ASSERT(!pParameterChainToken,"This shouldn't have been extracted");
-		cppUngetString(vStringValue (pInfo->replacement),vStringLength (pInfo->replacement));
-		CXX_DEBUG_LEAVE_TEXT("Applied simple replacement %s",vStringValue (pInfo->replacement));
-		return;
 	}
 
-	const char * c = vStringValue (pInfo->replacement);
-	const char * e = c + vStringLength (pInfo->replacement);
-
-	vString * pReplacement = vStringNew();
-
-	const char * b = c;
-
 	CXXTokenChain * pParameters = NULL;
-
-	while(c < e)
-	{
-		if(*c == '$')
+	const char ** aParameters = NULL;
+	int iParameterCount = 0;
+
+	if(pInfo->hasParameterList && pParameterChainToken && (pParameterChainToken->pChain->iCount >= 3))
+	{
+		// kill parenthesis
+		cxxTokenChainDestroyFirst(pParameterChainToken->pChain);
+		cxxTokenChainDestroyLast(pParameterChainToken->pChain);
+
+		pParameters = cxxTokenChainSplitOnComma(
+				pParameterChainToken->pChain
+			);
+		
+		aParameters = (const char **)eMalloc(sizeof(const char *) * pParameters->iCount);
+		CXXToken * pParam = cxxTokenChainFirst(pParameters);
+		while(pParam)
 		{
-			if(c > b)
-				vStringNCatS(pReplacement,b,c-b);
-			c++;
-			if(c >= e)
-			{
-				// stray $
-				vStringPut(pReplacement,'$');
-				break;
-			}
-
-			int idx;
-
-			if(*c == '*')
-			{
-				idx = -1; // all parameters
-			} else {
-				if((*c < '0') && (*c > '9'))
-				{
-					vStringPut(pReplacement,'$');
-					b = c;
-					continue;
-				}
-				idx = *c - '0';
-			}
-
-
-			if(!pParameters)
-			{
-				if(pParameterChainToken && (pParameterChainToken->pChain->iCount >= 3))
-				{
-					// kill parenthesis
-					cxxTokenChainDestroyFirst(pParameterChainToken->pChain);
-					cxxTokenChainDestroyLast(pParameterChainToken->pChain);
-
-					pParameters = cxxTokenChainSplitOnComma(
-							pParameterChainToken->pChain
-						);
-
-					cxxTokenChainCondense(pParameterChainToken->pChain,0);
-
-					CXX_DEBUG_PRINT(
-							"Parameter chain is '%s'",
-							vStringValue(cxxTokenChainFirst(pParameterChainToken->pChain)->pszWord)
-						);
-				}
-			}
-
-			if(pParameters)
-			{
-				if(idx >= 0)
-				{
-					CXX_DEBUG_PRINT("Applying $%d",idx);
-
-					CXXToken * pParameter = cxxTokenChainAt(pParameters,idx);
-					if(pParameter)
-						vStringCat(pReplacement,pParameter->pszWord);
-				} else {
-					if(pParameterChainToken)
-					{
-						// Note that we have condensed the chain above!
-						CXXToken * pAll = cxxTokenChainFirst(
-								pParameterChainToken->pChain
-							);
-						if(pAll)
-							vStringCat(pReplacement,pAll->pszWord);
-					}
-				}
-
-			}
-
-			c++;
-			b = c;
-		} else {
-			c++;
+			aParameters[iParameterCount] = vStringValue(pParam->pszWord);
```

Comment:
probably a dumb question from not knowing this new C++ parser enough yet, but is `pszWord` always set, for all tokens?  And is it already "compressed" to work with params like function calls and stuff?